src/Spatial/Internals/ImmutableList.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace MathNet.Spatial.Internals
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    internal sealed class ImmutableList<T> : IEnumerable<T>
```

Comment:
Not sure about adding `IEnumerable<T>` here as it invites use of LINQ.

Reply:
linq in intself is not evil - sometimes its even efficient, and with the new benchmarking we should catch poor implementations better

Reply:
ok, keeping it for now then, can always be refactored later.